src/components/settings/PresetButtons.svelte

```diff
@@ -0,0 +1,136 @@
+<script lang="ts">
+  import {
+    captionDuration,
+    captionFontSize,
+    captionWidth,
+    captionLeft,
+    captionTop,
+    videoSideSetting,
+    chatZoom,
+    livetlFontSize,
+    textDirection,
+    chatSplit,
+    activePreset,
+    presets,
+    videoPanelSize,
+    chatSize,
+    keyboardShortcuts,
+    doTranslatorMode,
+    doAutoPrefix,
+    autoPrefixTag,
+    macroTrigger,
+    macros
+  } from '../../js/store';
+  import { importStores } from '../../js/storage';
+  import Button from 'smelte/src/components/Button';
+
+  let presetCount = $presets.length;
+  function getPresetArray(presetCount: number) {
+    const returnArr = [];
+    for (let i = 1; i < Math.floor(presetCount / 2) + 1; i++) {
+      returnArr.push(i);
+    }
+    if (presetCount % 2 !== 0) { returnArr.push(presetCount / 2 + 0.5); }
+    return returnArr;
+  }
+
+  let isDeleting = false;
+  let active = $activePreset;
+  $: activePreset.set(active);
+
+  async function updatePreset(number: number) {
+    if (number !== active) {
+      importStores(JSON.stringify($presets[number - 1]));
+      active = number;
+      return;
+    }
+
+    const presetData = $presets;
+    presetData[number - 1] = {
+      captionDuration: $captionDuration,
+      captionFontSize: $captionFontSize,
+      captionWidth: $captionWidth,
+      captionLeft: $captionLeft,
+      captionTop: $captionTop,
+
+      videoSideSetting: $videoSideSetting,
+      videoPanelSize: $videoPanelSize,
+
+      chatZoom: $chatZoom,
+      chatSize: $chatSize,
+      chatSplit: $chatSplit,
+
+      textDirection: $textDirection,
+      livetlFontSize: $livetlFontSize,
+
+      keyboardShortcuts: $keyboardShortcuts,
+
+      doTranslatorMode: $doTranslatorMode,
+      doAutoPrefix: $doAutoPrefix,
+      autoPrefixTag: $autoPrefixTag,
+      macroTrigger: $macroTrigger,
+      macros: $macros
```

Comment:
There seem to be a lot of stores here in this massive object. It might be better to export an array of these stores from `store.js` and then do
```js
// presetStores is an array with all the stores that will be persisted in presets
presetData[number - 1] = Object.fromEntries(presetStores.map(store => [store.name, store.get()]));
```
and do something similar for the default value object you make later.